aten/src/ATen/SparseGCSTensorImpl.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/SparseCOOTensorImpl.h>
+#include <c10/core/TensorImpl.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+  
+// TODO: since many methods in SparseCOOTensorImpl can be used by GCS sparse tensor directly
+// we probably should have some superclass between TensorImpl and GCSTensorImpl that is
+// shared between COO and GCS tensors.
+struct CAFFE2_API SparseGCSTensorImpl : public TensorImpl {
+  Tensor pointers_;
+  Tensor indices_;
+  Tensor values_;
+  Tensor reduction_;
+
+  // Data for making index conversion operations faster.
+
+  // strides of the first half of the split dimensions.
+  std::vector<int> strides0_;
+  // strides of the second half of the split dimensions.
+  std::vector<int> strides1_;
+  // dims of the first half of the split dimensions.
+  std::vector<int> dims0_;
+  // dims of the second half of the split dimensions.
+  std::vector<int> dims1_;
```

Comment:
These are pretty weird and I as I read the rest I will be trying to understand why you actually need this.

Reply:
`std::vector` members are not free; each of them will be a dynamic allocation (a mini-tensor); so in effect, you are representing a GCS tensor with EIGHT arrays and that sets off tons of alarm bells

Reply:
I read the RFC and it looks like this is the dimension split generalization. I commented more about it at https://github.com/pytorch/rfcs/pull/6/files#r541043064

Reply:
I'm sorry I cannot see your comment anymore. Can you please post it one more time?

Do you think the storage should be done inside one contiguous array that is demarcated at various indices?